File: 303-range-sum-query-immutable/range-sum-query-immutable.java
class NumArray {
    int[] result;
    public NumArray(int[] nums) {
        result = new int[nums.length];
        int prefixSum = 0;
        for(int i = 0; i < result.length; i++) {
            prefixSum += nums[i];
            result[i] = prefixSum;
        }
        System.out.println(Arrays.toString(result));
    }
    
    public int sumRange(int left, int right) {
        if(left == 0) {
            return result[right];
        } else {
            return result[right] - result[left-1];
        }
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: 303-range-sum-query-immutable/range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefixArray = nums
        sum = self.prefixArray[0]
        for i in range(1, len(self.prefixArray)):
            sum += self.prefixArray[i]
            self.prefixArray[i] = sum


    def sumRange(self, left: int, right: int) -> int:
        if left == 0:
            return self.prefixArray[right]
        else:
            return self.prefixArray[right] - self.prefixArray[left - 1]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)